_posts/2020-03-15-learning-to-share-accomplishments-2.md

```diff
@@ -0,0 +1,214 @@
+  in a separate audit table, both making saves and lookups easy.
+* We needed a hierarchy of who could add, edit, or delete particular sets of data. I came up with the idea of tying in 
+  our system with Active Directory so we could actually look up the management ladder and allow people to edit the 
+  people's entries that were under them. If they moved teams, left the company, etc., they lost access as it 
+  was in AD already, and required no special login or management tables in our product.
+* We needed a way to create and run reports. We learned about a tool other teams used but realized on a short time 
+  frame we probably wouldn't integrate it well so we built our own. I designed the structure of how people would 
+  design reports, and then later with another team member, came up with a structure for how it would be stored in the 
+  database.
+* Reports were VERY slow in the beginning. I developed a caching system that would pre-calculate all calculated 
+  metrics, and store those values as they were entered. Using the cached data took reports from a minute or longer to 
+  load to about a second or less to load. 
+* The other developers shied away from creating PDFs. I found a library and implemented an export of a report to PDF 
+  files.
+* Not only did we get a minimal viable product out in three months, but the next group of five hired after us took over 
+  the project to finish our features. In addition, other departments in the bank saw our project and loved the open 
+  design of it and wanted to adopt it for their own use.
+
+### My Successes While Interning For Two Summers
+
+I interned with a electronic document management company that loved me so much they brought me back the next summer. I 
+was on different teams each time.
+
+**What did I Do?**
+
+* The first summer, I was handed a tiny bit of code that was supposed to build out a set of sample documents to test 
+  in a report. After two months, I had expanded that from that tiny bit to an entire testing framework that would create 
+  documents and test them against the business intelligence reports automatically. This reduced the work of our QA 
+  person about 50% because she didn't have to manually test reports.
+* I added internationalization to the the business intelligence installer application.
+* The second summer, I was given a C# program to add features to. I had not learned C# before. I added the features 
+  requested, but also discovered some quirks in how it ran on particular clients' machines. I also fixed those 
+  additional bugs.
+* I added a variety of features to a large JavaScript app that used jQuery (which I had never used at the time).
+* During the second summer, I helped fellow interns not only understand the products they were working on, but often 
+  went over and paired with them on debugging sessions to find errors and help them "rubber duck" on tasks. I 
+  helped out 5 interns that summer.
```

Comment:
I also had an "so you were here last year, here, take these kiddos" experience <3

Reply:
I'm glad I could step in and help get them up to speed faster!